UI/Utilities/ObservableMap.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HuskyRobotics.Utilities
+{
+    public class ObservableMap<T> : INotifyPropertyChanged
+    {
+        private IDictionary<String, MeasuredValue<T>> dictionary = new Dictionary<String, MeasuredValue<T>>();
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        /// <summary>
+        /// Gets the string representation of the value at the given dictionary key. Designed for bindings
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public MeasuredValue<T> this[string key] {
+            get => dictionary[key];
+        }
+
+        public bool ContainsKey(string key) => dictionary.ContainsKey(key);
+
+        public T GetValue(string key) => dictionary[key].Value;
+
+        public void PutValue(string key, T value)
+        {
+            if (dictionary.ContainsKey(key))
+            {
+                dictionary[key].Value = value;
+            } else
+            {
+                dictionary[key] = new MeasuredValue<T>(value);
+                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs("Count"));

```

Comment:
Is this "count" parameter going to be used for anything? If not is it needed?

Reply:
Not that I'm aware of. I included it for completeness. Is it worth removing?